homeassistant/components/sensor/openhardwaremonitor.py

```diff
@@ -0,0 +1,192 @@
+            values = array[path_number]
+
+            if path_index == len(self.path) - 1:
+                self.value = values[OHM_VALUE].split(' ')[0]
+                _attributes.update({
+                    'name': values[OHM_NAME],
+                    STATE_MIN_VALUE: values[OHM_MIN].split(' ')[0],
+                    STATE_MAX_VALUE: values[OHM_MAX].split(' ')[0]
+                })
+
+                self.attributes = _attributes
+                return
+            else:
+                array = array[path_number][OHM_CHILDREN]
+                _attributes.update({
+                    'level_%s' % path_index: values[OHM_NAME]
+                })
+
+
+class OpenHardwareMonitorData(object):
+    """Class used to pull data from OHM and create sensors."""
+
+    def __init__(self, config, hass):
+        """Initialize the Open Hardware Monitor data-handler."""
+        self.data = None
+        self._config = config
+        self._hass = hass
+        self.devices = []
+        self.initialize(utcnow())
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Hit by the timer with the configured interval."""
+        if self.data is None:
+            self.initialize(utcnow())
+        else:
+            self.refresh()
+
+    def refresh(self):
+        """Download and parse JSON from OHM."""
+        data_url = "http://%s:%d/data.json" % (
+            self._config.get(CONF_HOST),
+            self._config.get(CONF_PORT))
+
+        try:
+            response = requests.get(data_url)
+            self.data = response.json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is OpenHardwareMonitor running?")
+
+    def schedule_retry(self):
+        """Schedule a retry in 30 seconds."""
+        _LOGGER.info("Retrying in 30 seconds")
+
+        async_track_point_in_utc_time(
+            self._hass, self.initialize, utcnow() + RETRY_INTERVAL)
+
+    def initialize(self, now):
+        """Initial parsing of the sensors and adding of devices."""
+        self.refresh()
+
+        if self.data is None:
+            self.schedule_retry()
+            return
+
+        self.devices = self.parse_children(self.data, [], [], [])
+
+    def parse_children(self, json, devices, path, names):
+        """Recursively loop through child objects, finding the values."""
+        result = devices.copy()
+
+        if len(json[OHM_CHILDREN]) > 0:
+            for child_index in range(0, len(json[OHM_CHILDREN])):
+                child_path = path.copy()
+                child_path.append(child_index)
+
+                child_names = names.copy()
+                if len(path) > 0:
+                    child_names.append(json[OHM_NAME])
+
+                obj = json[OHM_CHILDREN][child_index]
+
+                added_devices = self.parse_children(
+                    obj, devices, child_path, child_names)
+
+                result = result + added_devices
+        else:
```

Comment:
Instead of an else, just put a `return` at the end of the if statement. Saves 1 indentation.